src/raftstore/store/worker/pd.rs

```diff
@@ -174,6 +182,57 @@ impl<T: PdClient> Runner<T> {
             error!("report split failed {:?}", e);
         }
     }
+
+    fn handle_validate_peer(&self, local_region: metapb::Region, peer: metapb::Peer) {
+        metric_incr!("pd.get_region");
+        match self.pd_client.get_region(local_region.get_start_key()) {
+            Ok(mut pd_region) => {
+                metric_incr!("pd.get_region");
+                if pd_region == local_region {
+                    // Local region info is as fresh as pd, which means
+                    // local peer is still in the region, let it be there.
+                    info!("peer {:?} is still valid because local region info is as fresh as pd",
+                          peer);
+                    return;
+                }
+                if !(pd_region.get_start_key() == local_region.get_start_key()) {
```

Comment:
> If node B is not removed out of the cluster by member change, at the end it would has the region range [0, 100), 
Why [0, 100)? B's origin range is [100, 200), how can B cover [0, 100) ? 

IMO, if we get a region which covers the search key, and the region ids are mismatched, we can think the old region is stale.